pyiron_continuum/fenics/wrappers.py

```diff
@@ -0,0 +1,333 @@
+        if self._as_fenics is None or len(args) > 0 or len(kwargs) > 0:
+            self.generate(*args, **kwargs)
+        return self._as_fenics
+
+    @property
+    def input_string(self):
+        return self.storage.input_string
+
+    @property
+    def kwargs(self):
+        return self.storage.kwargs
+
+    def __str__(self):
+        return self.input_string
+
+
+class Condition(FenicsWrapper):
+    """
+    Creates a function that tests the spatial value (canonically `x` in fenics) and whether that point is `on_boundary`.
+    The input string defines the condition to be evaluated, and all kwargs are exec'd beforehand.
+    """
+    @property
+    def _parser(self):
+        return StringInputParser(known_elements=["x", near])
+
+    def _generate(self):
+        for k, v in self.kwargs.items():
+            exec(f'{k} = {v}')
+
+        def boundary_func(x, on_boundary):
+            try:
+                return on_boundary and eval(self.input_string or "True")  # Allow for empty condition strings
+            except Exception as err_msg:
+                print(err_msg)
+
+        return boundary_func
+
+
+class Value(FenicsWrapper):
+    """
+    Creates a fenics `Constant` or `Expression` depending whether `degree` is included in the kwargs.
+
+    The fenics class ("Constant" or "Expression") can be omitted from the input string, and all kwargs are passed
+    directly to the fenics class itself.
+    """
+    @property
+    def _parser(self) -> StringInputParser:
+        return StringInputParser(known_elements=["x", "exp", "pow"])
+
+    def _generate(self):
+        if 'degree' in self.kwargs.keys():
+            return self._generate_expression(self.input_string, **self.kwargs)
+        else:
+            return self._generate_constant(self.input_string, **self.kwargs)
+
+    @staticmethod
+    def _generate_expression(value_tuple, **kwargs):
+        return Expression(eval(value_tuple), **kwargs)
+
+    @staticmethod
+    def _generate_constant(value_tuple, **kwargs):
+        return Constant(eval(value_tuple), **kwargs)
+
+
+class DirichletBC(HasStorage):
+    def __init__(self, value: Value = None, condition: Condition = None):
+        super().__init__()
+        self.storage.value = value
+        self.storage.condition = condition
+
+    @property
+    def value(self):
+        return self.storage.value
+
+    @property
+    def condition(self):
+        return self.storage.condition
+
+    def __call__(self, function_space):
+        return FEN.DirichletBC(function_space, self.value(), self.condition())
+
+    def __str__(self):
+        return str(self.value) + ", " + str(self.condition)
+
+
+class Mesh(FenicsWrapper):
+    """
+    Creates a spatial mesh.
+
+    The input string should include all fenics code to be evaluated, including the class to be called, while all kwargs
+    are exec'd beforehand.
+    """
+    @property
+    def _parser(self):
+        return StringInputParser(known_elements=[
```

Comment:
It looks like pyiron already knows all the possibilities that FEniCS offers. Shouldn't we expose them more?

Reply:
I'm not at all sure this is comprehensive. But yeah, I agree this attribute should be better exposed!

Reply:
I'll add this next time I'm working